app/test/consistent_apphash_test.go

```diff
@@ -35,66 +39,119 @@ import (
+type blobTx struct {
 	author    string
 	blobs     []*share.Blob
 	txOptions []user.TxOption
 }
 
-// TestConsistentAppHash executes all state machine messages, generates an app hash,
+type (
+	encodedSdkMessages func(*testing.T, []sdk.AccAddress, []stakingtypes.Validator, *app.App, *user.Signer, *user.Signer) ([][]byte, [][]byte, [][]byte)
+	encodedBlobTxs     func(*testing.T, *user.Signer, []sdk.AccAddress) []byte
+)
+
+type appHashTest struct {
+	name               string
+	version            uint64
+	encodedSdkMessages encodedSdkMessages
+	encodedBlobTxs     encodedBlobTxs
+	expectedDataRoot   []byte
+	expectedAppHash    []byte
+}
+
+// TestConsistentAppHash executes all state machine messages on all app versions, generates an app hash,
 // and compares it against a previously generated hash from the same set of transactions.
 // App hashes across different commits should be consistent.
 func TestConsistentAppHash(t *testing.T) {
-	// Expected app hash produced by v1.x - https://github.com/celestiaorg/celestia-app/blob/v1.x/app/consistent_apphash_test.go
-	expectedAppHash := []byte{84, 216, 210, 48, 113, 204, 234, 21, 150, 236, 97, 87, 242, 184, 45, 248, 116, 127, 49, 88, 134, 197, 202, 125, 44, 210, 67, 144, 107, 51, 145, 65}
-	expectedDataRoot := []byte{100, 59, 112, 241, 238, 49, 50, 64, 105, 90, 209, 211, 49, 254, 211, 83, 133, 88, 5, 89, 221, 116, 141, 72, 33, 110, 16, 78, 5, 48, 118, 72}
-
-	// Initialize testApp
-	testApp := testutil.NewTestApp()
-	enc := encoding.MakeConfig(app.ModuleEncodingRegisters...)
-
-	// Create deterministic keys
-	kr, pubKeys := deterministicKeyRing(enc.Codec)
-
-	// Apply genesis state to the app.
-	valKeyRing, _, err := testutil.SetupDeterministicGenesisState(testApp, pubKeys, 20_000_000_000, app.DefaultInitialConsensusParams())
-	require.NoError(t, err)
+	tc := []appHashTest{
+		{
+			name:               "execute sdk messages and blob tx on v1",
+			version:            v1.Version,
+			encodedSdkMessages: encodedSdkMessagesV1,
+			encodedBlobTxs:     createEncodedBlobTx,
+			expectedDataRoot:   []byte{100, 59, 112, 241, 238, 49, 50, 64, 105, 90, 209, 211, 49, 254, 211, 83, 133, 88, 5, 89, 221, 116, 141, 72, 33, 110, 16, 78, 5, 48, 118, 72},
+			// Expected app hash produced by v1.x - https://github.com/celestiaorg/celestia-app/blob/v1.x/app/consistent_apphash_test.go
+			expectedAppHash: []byte{84, 216, 210, 48, 113, 204, 234, 21, 150, 236, 97, 87, 242, 184, 45, 248, 116, 127, 49, 88, 134, 197, 202, 125, 44, 210, 67, 144, 107, 51, 145, 65},
+		},
+		{
+			name:    "execute sdk messages and blob tx on v2",
+			version: v2.Version,
+			encodedSdkMessages: func(t *testing.T, accountAddresses []sdk.AccAddress, genValidators []stakingtypes.Validator, testApp *app.App, signer *user.Signer, valSigner *user.Signer) ([][]byte, [][]byte, [][]byte) {
+				firstBlockEncodedTxs, secondBlockEncodedTxs, thirdBlockEncodedTxs := encodedSdkMessagesV1(t, accountAddresses, genValidators, testApp, signer, valSigner)
```

Comment:
v2 removed x/blobstream so I think the blobstream transactions need to be removed the test case on v2. Alternatively if they aren't removed, maybe this test case should verify that x/blobstream messages are no-ops on v2.

Reply:
I think those messages are supposed to be rejected on v2. I'll investigate.

Reply:
It looks like the blobstream messages were never part of the v1 test suite anyway so I think doing it like this is fine

Reply:
If the v1 test doesn't have blobstream messages, is it missing any other messages? Do we have a GH issue to make this test exhaustive of all message types supported by each app version?

Reply:
it covers most of it now. the only message we don't execute is `MsgSubmitEvidence` and IBC messages that require a different setup. Initially, we wanted to cover as much as possible and we're now extending to cover IBC messages too.